internal/controller/telemetry/route_controller.go

```diff
@@ -358,13 +424,87 @@ func (r *RouteReconciler) getLogsourceNamespaceNamesForTenant(ctx context.Contex
 	}
 
 	namespaceNames := make([]string, len(namespaces))
-
 	for i, namespace := range namespaces {
 		namespaceNames[i] = namespace.Name
 	}
 
 	return namespaceNames, nil
+}
 
+func (r *RouteReconciler) getBridges(ctx context.Context, listOpts *client.ListOptions) ([]v1alpha1.Bridge, error) {
+	var bridges v1alpha1.BridgeList
+	if err := r.Client.List(ctx, &bridges, listOpts); client.IgnoreNotFound(err) != nil {
+		return nil, err
+	}
+
+	return bridges.Items, nil
+}
+
+func (r *RouteReconciler) getBridgesForTenant(ctx context.Context, tenantName string) (bridgesOwned []v1alpha1.Bridge, err error) {
+	listOpts := &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(bridgeSourceTenantReferenceField, tenantName),
+	}
+	sourceBridge, err := r.getBridges(ctx, listOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	listOpts = &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(bridgeTargetTenantReferenceField, tenantName),
+	}
+	targetBridge, err := r.getBridges(ctx, listOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	bridges := append(sourceBridge, targetBridge...)
+	for _, bridge := range bridges {
+		if bridge.Spec.SourceTenant == tenantName || bridge.Spec.TargetTenant == tenantName {
+			bridgesOwned = append(bridgesOwned, bridge)
+		}
+	}
+
+	return
+}
+
+func (r *RouteReconciler) getTenants(ctx context.Context, listOpts *client.ListOptions) ([]v1alpha1.Tenant, error) {
+	var tenants v1alpha1.TenantList
+	if err := r.Client.List(ctx, &tenants, listOpts); client.IgnoreNotFound(err) != nil {
+		return nil, err
+	}
+
+	return tenants.Items, nil
+}
+
+func (r *RouteReconciler) checkBridgeConnections(ctx context.Context, bridge *v1alpha1.Bridge) error {
+	listOpts := &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(tenantNameField, bridge.Spec.SourceTenant),
+	}
+	sourceTenant, err := r.getTenants(ctx, listOpts)
+	if err != nil {
+		return err
+	}
+
+	listOpts = &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(tenantNameField, bridge.Spec.TargetTenant),
+	}
+	targetTenant, err := r.getTenants(ctx, listOpts)
+	if err != nil {
+		return err
+	}
+
+	// Granular check to ensure that the tenants are correctly assigned to the bridge
+	tenants := append(sourceTenant, targetTenant...)
+	if len(tenants) != 2 {
+		return errors.Errorf("bridge (%s) has invalid tenants", bridge.Name)
+	}
+
+	if !(tenants[0].Name == bridge.Spec.SourceTenant && tenants[1].Name == bridge.Spec.TargetTenant) ||
+		(tenants[0].Name == bridge.Spec.TargetTenant && tenants[1].Name == bridge.Spec.SourceTenant) {
```

Comment:
How about just this?
```suggestion
	if (tenants[0].Name != bridge.Spec.SourceTenant || tenants[1].Name != bridge.Spec.TargetTenant)
```

I think the second case (_reversed order_) is covered by the first !(_correct order_). Let me know if I missed / misunderstood something though.

Reply:
Yeah I'm also quite unhappy with this check let me come up with something better.